aiida/daemon/client.py

```diff
@@ -147,20 +152,20 @@ def get_circus_socket_directory(self):
         """
         if self.is_daemon_running:
             try:
-                return open(self.circus_socket_file, 'r').read().strip()
+                return io.open(self.circus_socket_file, 'r', encoding='utf8').read().strip()
             except (ValueError, IOError):
                 raise RuntimeError('daemon is running so sockets file should have been there but could not read it')
         else:
 
             # The SOCKET_DIRECTORY is already set, a temporary directory was already created and the same should be used
-            if self._SOCKET_DIRECTORY is not None:
-                return self._SOCKET_DIRECTORY
+            if self._socket_directory is not None:
+                return self._socket_directory
 
             socket_dir_path = tempfile.mkdtemp()
-            with open(self.circus_socket_file, 'w') as handle:
-                handle.write(socket_dir_path)
+            with io.open(self.circus_socket_file, 'w', encoding='utf8') as fhandle:
+                fhandle.write(six.text_type(socket_dir_path))
```

Comment:
My preference would be a separate codepath for Python 2:

```python2
if six.PY2:
  ...
else:
  ...
```

This way it would be clearer on what transformation is needed once we drop Python 2

Reply:
Resolved by /utils/json.py replacement.